### DISCUSSION TITLE
How to access ECS data from render graph nodes?

### DISCUSSION THREAD

**Discussion by KislyjKisel** [bevyengine/bevy#5030] [Q&A]
After extracting/preparing ECS data, what should I do to pass it to render graph node? Using `&World` to retrieve data is only possible when using resources or "manual" functions, but they require an entity. In sources I've found the entity was taken from an input slot connected to `VIEW_ENTITY`. I assume for custom entities other outputs are needed?

**Comment by djeedai** [comment]
Essentially you need to inject the data you need from the render node into the render world, one way or another, during the extract phase. That means either via a resource or via components on entities. Then you can read it back like you'd do for a normal system since the prepare and queue phases access the render world by default, not the app one. So just make a query. One subtlety here is the use of lifetimeless accessors like `SQuery` instead of `Query` but although I have an intuition about it I don't understand it enough to give a good explanation of it.
